ir/module.go

```diff
@@ -187,6 +188,160 @@ func (m *Module) String() string {
 	return buf.String()
 }
 
+type ioWriteWrapping struct {
+	size int
+	err  error
+}
+
+func (i *ioWriteWrapping) Fprintln(w io.Writer, a ...interface{}) {
+	if i.err != nil {
+		n, err := fmt.Fprintln(w, a)
+		i.size += n
+		i.err = err
+	}
+}
+func (i *ioWriteWrapping) Fprint(w io.Writer, a ...interface{}) {
+	if i.err != nil {
+		n, err := fmt.Fprint(w, a)
+		i.size += n
+		i.err = err
+	}
+}
+func (i *ioWriteWrapping) Fprintf(w io.Writer, format string, a ...interface{}) {
+	if i.err != nil {
+		n, err := fmt.Fprintf(w, format, a)
+		i.size += n
+		i.err = err
+	}
+}
+
+// WriteTo write the string representation of the module in LLVM IR assembly
+// syntax to output.
+func (m *Module) WriteTo(output io.Writer) (size int, err error) {
+	writeWrapping := &ioWriteWrapping{size: 0, err: nil}
+	// Assign metadata IDs.
+	if err := m.AssignMetadataIDs(); err != nil {
+		panic(fmt.Errorf("unable to assign metadata IDs of module; %v", err))
+	}
+	// Source filename.
+	if len(m.SourceFilename) > 0 {
+		// 'source_filename' '=' Name=StringLit
+		writeWrapping.Fprintf(output, "source_filename = %s\n", quote(m.SourceFilename))
+	}
+	// Data layout.
+	if len(m.DataLayout) > 0 {
+		// 'target' 'datalayout' '=' DataLayout=StringLit
+		writeWrapping.Fprintf(output, "target datalayout = %s\n", quote(m.DataLayout))
+	}
+	// Target triple.
+	if len(m.TargetTriple) > 0 {
+		// 'target' 'triple' '=' TargetTriple=StringLit
+		writeWrapping.Fprintf(output, "target triple = %s\n", quote(m.TargetTriple))
+	}
+	// Module-level inline assembly.
+	if len(m.ModuleAsms) > 0 {
```

Comment:
Just noticed that the `&& buf.Len() > 0`  check was removed from these lines. The equivalent line would be `&& writeWrapping.size > 0`.